_includes/promos/10tvpromo.html

```diff
@@ -81,26 +123,81 @@ <h4><br/>Offer valid until May 1st, 2020<br />To submit your business, fill out
           charLeft.innerText = currentLength;
       });
 
-            (function() {
+            ( function() {
             // We are currently not using babel. ES 5 syntax
             'use strict';
-            window.addEventListener('load', function() {
+            window.addEventListener('load', async function() {
                 // Fetch all the forms we want to apply custom Bootstrap validation styles to
                 var forms = document.getElementsByClassName('needs-validation');
                 if (forms.length > 0) {
                     for (var form of forms) {
-                        form.addEventListener('submit', function(e) {
+                        form.addEventListener('submit', async function(e) {
+                            console.log("Submitting");
                             if (!form.checkValidity()) {
+                                errorText.style.color = 'white';
+                                submitText.style.color = 'white';
                                 e.stopPropagation();
-                                
+                                e.preventDefault();
+                                console.log("FORM NOT VALID");
+                                form.classList.add('was-validated');
                             } else {
-                            submitText.style.color = 'black';
-                            submitButton.className = 'btn btn-success';
-                            submitButton.innerHTML = '&check;';
-                            submitButton.disabled = true;
+                                e.preventDefault();
+                                console.log("COMMENCING CHECK");
+                                myFunction()
+                                .catch(stuff => {
```

Comment:
mismatching .catch and `async`.

On async and await functions, you want to use a pattern of `try ... catch`.

Reply:
Hi Sean! I will be talking to Mark tomorrow but there might be a possibility that we don't need the async/await at all if we call a lambda-oriented api that can handle the post for us and hopefully check for duplicate records. 

In any other case, I would definitely agree with you. It worked locally on my machine with the catch, .then, asyncs, awaits and Promises with good ol' jquery but instilling try/catch in all of my functionality will be much more helpful.